#Pick compiler and flags from environmental variables and make libtriangle.a from triangle.o
# Aug 04th 2016 by B. Hajgato (Free Univeristy Brussels - VUB)
--- makefile.orig	2005-07-28 02:22:01.000000000 +0200
+++ makefile	2016-08-04 14:29:37.360139408 +0200
@@ -18,7 +18,7 @@
 
 # CC should be set to the name of your favorite C compiler.
 
-CC = cc
+CC ?= cc
 
 # CSWITCHES is a list of all switches passed to the C compiler.  I strongly
 #   recommend using the best level of optimization.  I also strongly
@@ -73,7 +73,7 @@
 #
 #   CSWITCHES = -O -DNO_TIMER -DLINUX -I/usr/X11R6/include -L/usr/X11R6/lib
 
-CSWITCHES = -O -DLINUX -I/usr/X11R6/include -L/usr/X11R6/lib
+CSWITCHES = $(CFLAGS) -DLINUX
 
 # TRILIBDEFS is a list of definitions used to compile an object code version
 #   of Triangle (triangle.o) to be called by another program.  The file
@@ -96,7 +96,7 @@
 
 all: $(BIN)triangle $(BIN)showme
 
-trilibrary: $(BIN)triangle.o $(BIN)tricall
+trilibrary: $(BIN)libtriangle.a $(BIN)tricall
 
 $(BIN)triangle: $(SRC)triangle.c
 	$(CC) $(CSWITCHES) -o $(BIN)triangle $(SRC)triangle.c -lm
@@ -109,6 +109,9 @@
 	$(CC) $(CSWITCHES) $(TRILIBDEFS) -c -o $(BIN)triangle.o \
 		$(SRC)triangle.c
 
+$(BIN)libtriangle.a: $(SRC)triangle.o
+	$(AR) cqs $(BIN)libtriangle.a triangle.o
+    
 $(BIN)showme: $(SRC)showme.c
 	$(CC) $(CSWITCHES) -o $(BIN)showme $(SRC)showme.c -lX11
 
